<launch>

    <!-- set some params for everybody -->
    <param name="/use_sim_time" value="true" />
    <param name="/simulation_active" value="true" />

    <rosparam file="$(find parallel_parameter_search)/config/master.yaml" command="load"/>
    <rosparam file="$(find parallel_parameter_search)/config/worker.yaml" command="load"/>

    <!-- load URDF -->
    <param name="robot_description" command="$(find xacro)/xacro $(find wolfgang_description)/urdf/wolfgang_basler.urdf.xacro" />
    <param name="urdf" value="$(find wolfgang_description)/urdf/wolfgang_basler.urdf" />
    <param name="robot_description_semantic" textfile="$(find wolfgang_moveit_config)/config/wolfgang.srdf" />
    <param name="robot_type_name" value="Wolfgang"/>
    
    <!-- Load updated joint limits (override information from URDF) -->
    <group ns="robot_description_planning">
        <rosparam command="load" file="$(find wolfgang_moveit_config)/config/joint_limits.yaml"/>
    </group>

    <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
    <group ns="robot_description_kinematics">
        <rosparam command="load" file="$(find wolfgang_moveit_config)/config/kinematics.yaml"/>
    </group>


    <!-- launch the trainMaster -->
    <node name="train_master" pkg="parallel_parameter_search" type="train_master.py" output="screen"/>


</launch>